gstd-meta/tests/meta_complex.rs

```diff
@@ -0,0 +1,173 @@
+// This file is part of Gear.
+
+// Copyright (C) 2021 Gear Technologies Inc.
+// SPDX-License-Identifier: GPL-3.0-or-later WITH Classpath-exception-2.0
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+#![no_std]
+
+// May be replaced with `use gstd_meta::*;`
+use gstd_meta::{meta, TypeInfo};
+
+extern crate alloc;
+use alloc::{boxed::Box, string::String, vec::Vec};
+
+#[allow(unused)]
```

Comment:
https://doc.rust-lang.org/reference/attributes/derive.html

you need to put `#[automatically_derived]` attribute to the macro output and remove all those `#[allow(unused)]` everywhere

Reply:
Usually, `#[automatically_derived]` is used while writing derive macro. It doesn't actually work in this case. Removed `#[allow(unused)]` from lib by using `pub` in structs and their fields.